tic-tac-toe/tictactoe.py

```diff
@@ -7,23 +7,69 @@ class Play(object):
     finished = False
 
     def __str__(self):
-        return self.game[0].__str__() + '\n' + self.game[1].__str__() + "\n" + self.game[2].__str__()
+        return "\n".join(map(str, self.game))
 
+    def check_finished(self):
+        result = False
+        for i in range(2):
+            if self.game[i][0] == self.game[i][1] == self.game[i][2] != '-':
+                result = self.game[i][0]
+            elif self.game[0][i] == self.game[1][i] == self.game[2][i] != '-':
+                result = self.game[i][0]
+        if self.game[0][0] == self.game[1][1] == self.game[2][2] != '-':
+            return self.game[0][0]
+        elif self.game[0][2] == self.game[1][1] == self.game[2][0] != '-':
+            return self.game[0][2]
+        elif map(str, self.game).__contains__("-"):
```

Comment:
`__contains__` is used by the `in` operator. You could also do:

`elif any("-" in row for row in self.game):`